File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        n = len(blocks)
        
        l = 0
        temp = 0
        for i in range(k):
            if blocks[i] == "W":
                temp += 1
        result = temp
        for i in range(k,n):
            if blocks[l] == "W":
                temp -= 1
            l += 1
            if blocks[i] == "W":
                temp += 1
            result = min(result,temp)
        return result


        

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        
        int n = blocks.length();
        int result = n;
        int temp = 0;
        for(int i = 0; i < n; i++){
            temp = 0;
            for(int j = i; j < n; j++){
                if(blocks.charAt(j) == 'W') temp++;
                if(j - i + 1 == k){
                    result = Math.min(result, temp);
                }
            }
        }
        return result;


    }
}